src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Metrics/LastValueAggregator.cs

```diff
@@ -0,0 +1,35 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace System.Diagnostics.Metrics
+{
+    internal class LastValue : Aggregator
+    {
+        private double? _lastValue;
+
+        public override void Update(double value)
+        {
+            _lastValue = value;
+        }
+
+        public override IAggregationStatistics Collect()
+        {
+            lock (this)
+            {
+                LastValueStatistics stats = new LastValueStatistics(_lastValue);
+                _lastValue = null;
+                return stats;
+            }
```

Comment:
Does `Update` need this lock too? If it isn't locking also, I'm not sure if we need this lock here. We could swap for an interlocked exchange or something.

Reply:
Yeah, you could do this one with an InterlockedExchange instead of the lock. I left it as lock because it was simple and I don't anticipate this code is on the hot path. The hot paths are InstrumentState.Update(), AggregatorStore.GetAggregator(), RateSumAggregator.Update() and Histogram.Update(). Everything else is unlikely to go above a few hundred invocations/sec.

Reply:
> Everything else is unlikely to go above a few hundred invocations/sec.

Which methods might be invoked several hundred times a second?  Ones that might allocate?

Reply:
> Which methods might be invoked several hundred times a second? Ones that might allocate?

Yeah, if someone requested to collect metrics once per second and enabled a few hundred metrics then the InstrumentState.Collect() path would be running a few hundred times per second. InstrumentState.Collect() invokes AggregatorStore.Collect() which in turn invokes Aggregator.Collect(). There are a few allocations down that path.

None of that is going to happen automatically, an engineer needed to run a diagnostic tool to turn it on, they needed to specify they wanted collections every second, and they needed to specify a bunch of metrics they wanted collected.